src/algorithm/core.py

```diff
@@ -707,50 +719,136 @@ def draw_marked_bubbles_with_evaluation_meta(
                 # is_field_incorrect = question_verdict == Verdict.NO_ANSWER_MATCH
                 # TODO: calculate question's scheme to get delta for all verdicts
                 #
-
+                question_verdict = evaluation_meta["questions_meta"][field_label]
+                if question_has_verdict:
+                    is_field_correct = question_verdict["question_verdict"].startswith(
+                        Verdict.ANSWER_MATCH
+                    )
+                    is_field_incorrect = (
+                        question_verdict["question_verdict"] == Verdict.NO_ANSWER_MATCH
+                    )
+                
                 for bubble_detection in field_bubble_means:
                     bubble = bubble_detection.item_reference
                     shifted_position = tuple(
                         bubble.get_shifted_position(field_block.shifts)
                     )
                     field_value = str(bubble.field_value)
-
+                    # logger.info(bubble)
                     # TODO: support for custom_labels verdicts too!
                     if question_has_verdict:
                         # TODO: make cases for colors for image_type == "COLORED" and box shapes for image_type == "GRAYSCALE"
                         # [marked, unmarked]  x [correct, incorrect, optional(if multiple answers)]
                         # update colors here
-                        pass
+                        if (
+                            field_value
+                            in evaluation_meta["questions_meta"][field_label][
+                                "expected_answer_string"
+                            ]
+                        ) or evaluation_meta["questions_meta"][field_label][
+                            "bonus_type"
+                        ]:
+
+                            ImageUtils.draw_box(
+                                marked_image,
+                                shifted_position,
+                                bubble_dimensions,
+                                CLR_BLACK,
+                                style="BOX_HOLLOW",
+                                thickness_factor=1,
+                            )
+                        if (
+                            evaluation_meta["questions_meta"][field_label]["bonus_type"]
+                            == "BONUS_ON_ATTEMPT"
+                        ):
+                            if (
+                                evaluation_meta["questions_meta"][field_label][
+                                    "marked_answer"
+                                ]
+                                == ""
```

Comment:
We can't rely on using "" since emptyVal can be configured by the user. Let's used `question_verdict == VerdictType.UNMARKED`